File: 2155-find-missing-observations/2155-find-missing-observations.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* missingRolls(int* rolls, int rollsSize, int mean, int n, int* returnSize) {
    int sum = 0;
    for (int i = 0; i < rollsSize; i++) sum += rolls[i];
    sum = (rollsSize + n) * mean - sum;
    if (sum > n * 6 || sum < n) {
        *returnSize = 0;
        return NULL;
    }
    int mod = sum % n, dice = sum / n;
    *returnSize = n;
    int* ans = (int*)malloc(n * sizeof(int));
    for (int i = 0; i < n; i++) {
        if (i < mod) ans[i] = dice + 1;
        else ans[i] = dice;
    }
    return ans;
}

File: 2155-find-missing-observations/2155-find-missing-observations.kt
class Solution {
    fun missingRolls(rolls: IntArray, mean: Int, n: Int): IntArray {
        val nSum : Int = (rolls.size + n) * mean - rolls.sum();
        if (nSum > n * 6 || nSum < n) return intArrayOf();
        val mod : Int = nSum % n;
        val ans : IntArray = IntArray(n) { nSum / n }.apply { repeat(mod) {
            it -> this[it]++;
        }};
        return ans;
    }
}